R/pspViz.R

```diff
@@ -0,0 +1,174 @@
+# figure 4, from the PSP paper, https://journals.asm.org/doi/epub/10.1128/msystems.00847-23
+# code from https://github.com/JRaviLab/psp_app/blob/main/psp_process.Rmd
+
+library(tidyverse)
+library(data.table)
+
+# source("/data/research/jravilab/molevol_scripts/R/cleanup.R")  # in package
+# source("/data/research/jravilab/molevol_scripts/R/summarize.R") # in package
+# source("/data/research/jravilab/molevol_scripts/R/plotting.R") # in package
+# source("/data/research/jravilab/molevol_scripts/R/reverse_operons.R") # in package
+# source("/data/research/jravilab/molevol_scripts/R/networks_domarch.R") # in package
+
+cln_colnames <- c("AccNum", "DomArch", "GenContext", "Length",  "GeneName",
+                  "Lineage", "Species", "GCA_ID", "GeneDesc",
+                  "DomArch.repeats", "GenContext.repeats")
+
+domains_rename <- read_delim("data/acc_files/domains_rename.txt",
+                             delim="\t", col_names=TRUE)
+
+# domains_ignore <- read_delim("data/acc_files/domains_ignore.txt",
+#                              delim="\t", col_names=T)
+
+domains_keep <- read_delim("data/acc_files/domains_keep.txt",
+                           delim="\t", col_names=T)
+
+query_domains <- read_delim("data/acc_files/query_domains.txt",
+                            delim="\t", col_names=T)
+lineage_rename <- read_tsv("data/acc_files/lineage_rename.txt",
+                           col_names=T)
+
+data_cleanup <- function(all_raw) {
+  # Cleanup Species
+  all_cln <- all_raw %>% cleanup_species(remove_empty=FALSE)
+  # Cleanup Gene Description
+  all_cln <- all_cln %>% cleanup_GeneDesc(column="GeneDesc") # Remove trailing '.'
+  # CLeanup Lineage
+  all_cln <- all_cln %>% cleanup_lineage(lins_rename=lineage_rename)
+
+  # Cleanup DomArch
+  # For the repeats
+  all_cln <- all_cln %>%
+    cleanup_domarch(domains_rename=domains_rename,
+                    domains_keep=NULL,   # filter applied to only ClustName for now.
+                    domains_ignore=NULL, #!! should check and remove soon!
+                    repeat2s=FALSE,
+                    remove_tails=F,      #new! check below if it works!
+                    remove_empty=F)      #new! needed?
```

Comment:
need to check with new function names | @the-mayer 
removing PSP-specific code
need logical checks | @epbrenner 

```suggestion
  all_cln <- all_cln %>%
    cleanup_domarch(domains_rename=domains_rename,
                    repeat2s=FALSE)
                    # remove_tails=F,      # FIXME # keep if it exists & works for DomArch
                    # remove_empty=F)      # FIXME # keep if it exists & works for DomArch

```